Week1/homework/app.js

```diff
@@ -4,8 +4,136 @@
   const bookTitles = [
     // Replace with your own book titles
     'harry_potter_chamber_secrets',
+    'the_count_of_monte_cristo',
+    'the_alchemist',
+    'footprint',
+    'revitalized_land',
+    'the_man_who_has_two_graves',
+    'home_land',
+    'the_last_theorem_of_fermat',
+    'the_old_man_and_the_sea',
+    'romeo_and_juliet',
   ];
 
   // Replace with your own code
-  console.log(bookTitles);
+  // 1.3
+  function createUlList(element) {
+    const ulList = document.createElement('ul');
+    ulList.setAttribute('id', 'bookTitlesList');
+    ulList.setAttribute('class', 'bookTitlesList');
+    for (let i = 0; i < element.length; i++) {
+      const liList = document.createElement('li');
+      liList.innerText = element[i];
+      liList.setAttribute('class', 'book');
+      ulList.appendChild(liList);
+    }
+    document.body.appendChild(ulList);
+  }
+  //  1.4
+  const bookInformationObject = {
+    harry_potter_chamber_secrets: {
+      title: bookTitles[0],
+      language: 'English',
+      author: 'J.K. Rowling',
+    },
+    the_count_of_monte_cristo: {
+      title: bookTitles[1],
+      language: 'English',
+      author: ' ‎Alexander Dumas',
+    },
+    the_alchemist: {
+      title: bookTitles[2],
+      language: 'English',
+      author: 'Paulo Coelho',
+    },
+    footprint: {
+      title: bookTitles[3],
+      language: 'Uyghur',
+      author: 'A. Otkur',
+    },
+    revitalized_land: {
+      title: bookTitles[4],
+      language: 'Uyghur',
+      author: 'A. Otkur',
+    },
+    the_man_who_has_two_graves: {
+      title: bookTitles[5],
+      language: 'Uyghur',
+      author: 'A. Turdi',
+    },
+    home_land: {
+      title: bookTitles[6],
+      language: 'Uyghur',
+      author: 'Z. Sabir',
+    },
+    the_last_theorem_of_fermat: {
+      title: bookTitles[7],
+      language: 'Chinese',
+      author: 'S. Singh',
+    },
+    the_old_man_and_the_sea: {
+      title: bookTitles[8],
+      language: 'English',
+      author: 'Ernest Hemingway',
+    },
+    romeo_and_juliet: {
+      title: bookTitles[9],
+      language: 'English',
+      author: 'William Shakespeare',
+    },
+  };
+  // 1.5
+  function DisplayBookInformation(element) {
+    const ul = document.createElement('ul');
+    ul.setAttribute('id', 'bookInformationList');
+    ul.setAttribute('class', 'bookInformationList');
+    // eslint-disable-next-line guard-for-in
+    for (const book in element) {
```

Comment:
Yash mentioned in class that we should avoid using `for in` loops as it may have unexpected results. Try and use Object.keys, Object.values or Object.entries instead.

Reply:
I will try it and push the changes.